focus-android/app/src/main/java/org/mozilla/focus/browser/integration/BrowserToolbarIntegration.kt

```diff
@@ -283,6 +295,67 @@ class BrowserToolbarIntegration(
+                            anchor = toolbar.findViewById<AppCompatEditText>(R.id.mozac_browser_toolbar_background),
+                            properties = CFRPopupProperties(
+                                popupWidth = 256.dp,
+                                popupAlignment = CFRPopup.PopupAlignment.BODY_TO_ANCHOR_START,
+                                popupBodyColors = listOf(
+                                    ContextCompat.getColor(
+                                        fragment.requireContext(),
+                                        R.color.cfr_pop_up_shape_end_color,
+                                    ),
+                                    ContextCompat.getColor(
+                                        fragment.requireContext(),
+                                        R.color.cfr_pop_up_shape_start_color,
+                                    ),
+                                ),
+                                dismissButtonColor = ContextCompat.getColor(
+                                    fragment.requireContext(),
+                                    R.color.cardview_light_background,
+                                ),
+                                popupVerticalOffset = 0.dp,
+                                indicatorArrowStartOffset = 10.dp,
+                            ),
+                            onDismiss = { onDismissCookieBannerCfr() },
+                            text = {
+                                val textCookieBannerCfr = stringResource(
+                                    id = R.string.cfr_for_cookie_banner,
+                                    LocalContext.current.getString(R.string.onboarding_short_app_name),
+                                )
+                                ClickableSubstringLink(
+                                    text = textCookieBannerCfr,
+                                    style = focusTypography.cfrCookieBannerTextStyle,
+                                    linkTextDecoration = TextDecoration.Underline,
+                                    clickableStartIndex = textCookieBannerCfr.indexOf(
+                                        LocalContext.current.getString(
+                                            R.string.cfr_for_cookie_banner_underline_settings,
+                                        ),
+                                    ),
+                                    clickableEndIndex = textCookieBannerCfr.length,
+                                    onClick = {
+                                        fragment.requireComponents.appStore.dispatch(
+                                            AppAction.OpenSettings(Screen.Settings.Page.CookieBanner),
+                                        )
+                                        onDismissCookieBannerCfr()
+                                    },
+                                )
+                            },
+                        ).apply {
+                            show()
```

Comment:
Should we call `stopObserverCookieBannerCfrChanges` after we show the CFR? As it should show once ,right?  Or should it show every time the setting is OFF?

Reply:
Done